.github/workflows/pull-request.yaml

```diff
@@ -1,38 +1,47 @@
 # This workflow will lint the PR's title and commit message.
-# https://github.com/CondeNast/conventional-pull-request-action
-# https://github.com/conventional-changelog/commitlint/blob/master/docs/reference-rules.md
-# https://github.com/CondeNast/conventional-pull-request-action/issues/15
 
 name: Pull Request
 on:
   pull_request:
     branches: [main, staging]
     types:
     - opened
+    - reopened
     - edited
     - synchronize
 permissions:
   contents: read
 
 jobs:
-  pr:
-    permissions:
-      pull-requests: read  # for CondeNast/conventional-pull-request-action to get specific PR & list commits
-    name: Lint PR title
+  conventional-commits:
+    name: Check PR #${{ github.event.pull_request.number }} title and commit messages
     runs-on: ubuntu-latest
     steps:
     - name: Harden Runner
       uses: step-security/harden-runner@74b568e8591fbb3115c70f3436a0c6b0909a8504
       with:
         egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
-
-    - name: Run conventional commits check
-      uses: CondeNast/conventional-pull-request-action@eef02ae6e88e4b462f39931113b1a75cc275beee
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Check out repository
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      with:
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08 # v4.1.0
       with:
-        commitlintRulesPath: ./commitlint.rules.js
-        commitTitleMatch: 'true'
+        python-version: '3.10'
+    - name: Create and activate virtual environment
+      run: |
+        make venv
+        source .venv/bin/activate
+      env:
+        PYTHON: python
+    - name: Install dependencies
+      run: make setup
```

Comment:
I also think it's useless to install the whole dependencies here. We could either install `commitizen` or add a new make target for conventional commits.

We should also consider if `harden-runner` is adding any value here. Since we don't use an external Actions workflow, I think we can disable it? We could do the same on other places where we don't use a third-party workflow?

Reply:
Commits https://github.com/jenstroeger/python-package-template/pull/263/commits/3d58166eb30baf037498bb95f7024ac342e0bb09 and https://github.com/jenstroeger/python-package-template/pull/263/commits/40890a364077b409fa3176447731772daccde871.